src/base/bittorrent/torrentimpl.cpp

```diff
@@ -2157,6 +2180,25 @@ bool TorrentImpl::setMetadata(const TorrentInfo &torrentInfo)
 #endif
 }
 
+Torrent::StopCondition TorrentImpl::stopCondition() const
+{
+    return m_stopCondition;
+}
+
+void TorrentImpl::setStopCondition(const StopCondition stopCondition)
+{
```

Comment:
you can still return early when `TorrentImpl::stopCondition() == stopCondition` to avoid following operations.